server/core/src/https/views/reset.rs

```diff
@@ -50,7 +50,72 @@ struct CredResetPartialView {
 #[derive(Serialize, Deserialize, Debug)]
 // Needs to be visible so axum can create this struct
 pub(crate) struct ResetTokenParam {
-    token: Option<String>
+    token: Option<String>,
+}
+
+
+#[derive(Template)]
+#[template(path = "cred_update/add_password_modal_partial.html")]
+struct AddPasswordModalPartial {
+    check_res: PwdCheckResult,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+enum PwdCheckResult {
+    Success,
+    Init,
+    Failure {
+        pwd_equal: bool,
+        warnings: Vec<PasswordFeedback>,
+    },
+}
+
+#[derive(Deserialize, Debug)]
+pub(crate) struct NewPassword {
+    new_password: String,
+    new_password_check: String,
+}
+
+pub(crate) async fn view_new_pwd(
+    State(state): State<ServerState>,
+    Extension(kopid): Extension<KOpId>,
+    HxRequest(_hx_request): HxRequest,
+    VerifiedClientInformation(_client_auth_info): VerifiedClientInformation,
+    jar: CookieJar,
+    Form(new_passwords): Form<NewPassword>,
+) -> axum::response::Result<Response> {
+    let cookie = jar.get(COOKIE_CU_SESSION_TOKEN);
+    return if let Some(cookie) = cookie {
+        let cu_session_token = cookie.value();
+        let cu_session_token = CUSessionToken { token: cu_session_token.into() };
+
+        let eq = new_passwords.new_password == new_passwords.new_password_check;
```

Comment:
`eq` is confusing here, and can save having to specify the name in line 99 later

```suggestion
        let pwd_equal = new_passwords.new_password == new_passwords.new_password_check;
```